android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/RepositoryServiceConnection.java

```diff
@@ -46,19 +45,30 @@ public RepositoryServiceConnection(Context context, ConnectionLifeCycle connecti
 
   @Override public void startConnection(final AppCoinsBillingStateListener listener) {
     this.listener = listener;
+    if (WalletUtils.hasWalletInstalled()) {
+      String packageName = WalletUtils.getBillingServicePackageName();
+      walletInstalledBehaviour(packageName);
+    } else {
+      walletNotInstalledBehaviour();
+    }
+  }
 
-    Intent serviceIntent = new Intent(BuildConfig.IAB_BIND_ACTION);
-    serviceIntent.setPackage(BuildConfig.IAB_BIND_PACKAGE);
+  private void walletNotInstalledBehaviour() {
+    hasWalletInstalled = false;
+    onServiceConnected(new ComponentName("", ""), new IBinderWalletNotInstalled());
+  }
 
+  private void walletInstalledBehaviour(String packageName) {
+    Intent serviceIntent = new Intent(BuildConfig.IAB_BIND_ACTION);
+    serviceIntent.setPackage(packageName);
 
     List<ResolveInfo> intentServices = context.getPackageManager()
         .queryIntentServices(serviceIntent, 0);
     if (intentServices != null && !intentServices.isEmpty()) {
       hasWalletInstalled = true;
```

Comment:
Does the `hasWalletInstalled` makes sense to be set on the behaviour methods, instead of being set when the if clause in fact asks if the wallet is installer (`WalletUtils.hasWalletInstalled`).

Also if we know that the wallet is installed is it possible for the intent service to be null or empty in the `walletInstalledBehaviour`? If so and since that is the main condition for the proceed with the behaviour the we should use the  `walletInstalledBehaviour` logic only instead of having one if clause in the `startConnection` and then a second in the `walletNotInstalledBehaviour`?